### PR TITLE
Add prepublish hook to generate the npm-shrinkwrap

### PR DESCRIPTION
**Pull request by gastonponti**
### Description

Fixed all the libraries that the cli is using at the moment of publishing it. 

### Tested

Published different versions of the cli under a different name, download the package and verified it.
Manually change the shrinkwrap to downgrade a version (package B), of a package required by a sub-package (package A) that is configurated to download the lastest version (of package B), and publish it again. The new published version downloaded the package B downgraded, instead of the lastest version

### Backwards compatibility

Yes

### CODE REVIEW

**Review by barbaraliau — COMMENTED** [review]
Doesn't removing `-dev` mean that `cli` won't track what's on the master branch for each of the changed packages? Should we instead pin the packages for each of the dependencies instead?

File: packages/cli/package.json [hunk 1]
```diff
@@ -25,20 +25,21 @@
...
-    "@celo/wallet-ledger": "1.0.1-dev",
+    "@celo/contractkit": "1.0.1",
+    "@celo/explorer": "1.0.1",
+    "@celo/identity": "1.0.1",
+    "@celo/governance": "1.0.1",
+    "@celo/utils": "1.0.1",
+    "@celo/wallet-local": "1.0.1",
+    "@celo/wallet-hsm-azure": "1.0.1",
+    "@celo/wallet-ledger": "1.0.1",
     "@ledgerhq/hw-transport-node-hid": "~5.11.0",
```
Comment (barbaraliau) [inline packages/cli/package.json]: Shouldn't we pin this and the oclif packages?

Comment (gastonponti) [reply packages/cli/package.json]: ohhhhhhhhh, my bad. I shouldn't push that. It was used to generate the packages that I tested.
Nice catch

Comment (gastonponti) [reply packages/cli/package.json]: Oh, you were pointing the other line, sorry (the @celo fixed versions were the wrong ones)

The idea, is that we keep the package.json as the same way we were handling it. We can still upgrade versions or having "non-fixed" ones.
The thing that maintains all the versions, is the `npm-shrinkwrap.json`
That `.json` is the one fixing everything.

(don't know how to reference lines in a comment from a different file)
Take a look on this
This line: https://github.com/celo-org/celo-monorepo/pull/6575/files#diff-25941123ee4dcd8d518962aa40b6b50a4eea5157e5494c135f8efc705fed752cR439
is basically pointing in the shrinkwrap that the library could eventually accept a version increment, but, check this other line:
https://github.com/celo-org/celo-monorepo/pull/6575/files#diff-25941123ee4dcd8d518962aa40b6b50a4eea5157e5494c135f8efc705fed752cR983
That is the line fixing it.

So, in the shrinkwrap file, you have what does every lib requires, but at the bottom, you have the list of the libs that were selected.

Probably, first it generates the file, and then searches for the minimum version acceptable for every lib to work, and then, it fixes that version.

Comment (barbaraliau) [reply packages/cli/package.json]: ohh i see. So shrinkwrap takes care of pinning the versions, and package.json looks the same.

Comment (barbaraliau) [reply packages/cli/package.json]: K this makes sense. Thanks for putting up the PR!

**Review by barbaraliau — APPROVED** [review]

**Pull request merged**